provider/oci/images_test.go

```diff
@@ -0,0 +1,393 @@
+	c.Assert(len(imageMap["centos7"][0].InstanceTypes), gc.Equals, 1)
+}
+
+func (i *imagesSuite) TestRefreshImageCacheStaleCache(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	compute := ocitesting.NewMockOCIComputeClient(ctrl)
+	defer ctrl.Finish()
+
+	compute.EXPECT().ListImages(gomock.Any(), gomock.Any()).Return(ociCore.ListImagesResponse{}, nil).Times(2)
+
+	imgCache, err := oci.RefreshImageCache(compute, &i.testCompartment)
+	c.Assert(err, gc.IsNil)
+	c.Assert(imgCache, gc.NotNil)
+
+	fromCache, err := oci.RefreshImageCache(compute, &i.testCompartment)
+	c.Assert(err, gc.IsNil)
+	c.Check(imgCache, gc.DeepEquals, fromCache)
+
+	now := time.Now()
+
+	// No need to check the value. gomock will assert if ListImages
+	// is not called twice
+	fromCache.SetLastRefresh(now.Add(-31 * time.Minute))
+	_, err = oci.RefreshImageCache(compute, &i.testCompartment)
+	c.Assert(err, gc.IsNil)
+}
+
+func (i *imagesSuite) TestRefreshImageCacheWithInvalidImage(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	compute := ocitesting.NewMockOCIComputeClient(ctrl)
+	defer ctrl.Finish()
+
+	listImageRequest, listImageResponse := makeListImageRequestResponse(
+		i.testCompartment, []map[string]string{
+			map[string]string{
+				"id":      "fakeUbuntu1",
+				"os":      "Canonical Ubuntu",
+				"name":    "Canonical-Ubuntu-14.04-2018.01.11-0",
+				"version": "14.04",
+			},
+			map[string]string{
+				"id":      "fake image id for bad image",
+				"os":      "CentOS",
+				"name":    "BadlyFormatedDisplayName_IshouldBeIgnored",
+				"version": "7",
+			},
+		})
+
+	shapesRequestUbuntu, shapesResponseUbuntu := makeShapesRequestResponse(
+		i.testCompartment, "fakeUbuntu1", []string{"VM.Standard2.1", "VM.Standard1.2"})
+
+	shapesRequestBadImage, shapesResponseBadImage := makeShapesRequestResponse(
+		i.testCompartment, "fake image id for bad image", []string{"VM.Standard1.2"})
+
+	gomock.InOrder(
+		compute.EXPECT().ListImages(context.Background(), listImageRequest).Return(listImageResponse, nil),
+		compute.EXPECT().ListShapes(context.Background(), shapesRequestUbuntu).Return(shapesResponseUbuntu, nil),
+		compute.EXPECT().ListShapes(context.Background(), shapesRequestBadImage).Return(shapesResponseBadImage, nil),
+	)
+
+	imgCache, err := oci.RefreshImageCache(compute, &i.testCompartment)
+	c.Assert(err, gc.IsNil)
+	c.Assert(imgCache, gc.NotNil)
+	c.Check(len(imgCache.ImageMap()), gc.Equals, 1)
+	imageMap := imgCache.ImageMap()
+
+	c.Check(imageMap["trusty"][0].Id, gc.Equals, "fakeUbuntu1")
+}
+
+func (i *imagesSuite) TestRefreshImageCacheWithAPIError(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	compute := ocitesting.NewMockOCIComputeClient(ctrl)
+	defer ctrl.Finish()
+
+	compute.EXPECT().ListImages(gomock.Any(), gomock.Any()).Return(ociCore.ListImagesResponse{}, errors.Errorf("awww snap!")).Times(1)
+	_, err := oci.RefreshImageCache(compute, &i.testCompartment)
+	c.Assert(err, gc.ErrorMatches, "awww snap!")
```

Comment:
Would we want to have some sort of errors.Annotate[f] in the call stack so we see something like:
err, gc.ErrorMatches, "while refreshing image cache: aww snap!")
?
I don't know what kind of errors we would be getting to know for sure, but some amount of annotation?